### PR TITLE
Get key public if requested

### PR DESCRIPTION
**Pull request by jschlyter**

### CODE REVIEW

File: src/cryptojwt/jwk.py [hunk 1]
```diff
@@ -407,6 +407,9 @@ def get_key(self, **kwargs):
         if not self.key:
             self.deserialize()
 
+        if not kwargs.get('private', True) and hasattr(self.key, 'public_key'):
```
Comment (jschlyter) [inline src/cryptojwt/jwk.py]: One could argue that requesting the public part of a symmetric key should fail.

**Comment by jschlyter** [comment]
ping @rohe

**Comment by rohe** [comment]
Sorry, I've been swamped with other stuff.
But I have 2 issues:
1) No tests !!
2) When you can't fulfil the request raise an Exception. This happens if you have a public key and is asked for a private key.

I'm also wondering whether it would be useful to have get_public_key, get_private_key too.
These would then for symmetric keys be NotImplemented or something.

**Comment by jschlyter** [comment]
What is the desired semantics for get_key(private=False) on a symmetric key? One could argue that a symmetric key has only a private key.

**Comment by rohe** [comment]
Ok, so what if we don't allow any arguments to get_key().
get_key() will return whatever key you have in there.
And then have get_private_key/get_public_key which are only implemented for asymmetric key types.

**Comment by jschlyter** [comment]
We remove the `private` flag to `get_key()`, we loose backwards compatibility with `jwkest` :-(

**Comment by rohe** [comment]
It's unfortunate that private has a default value :-( in jwkest.
Well, let's keep private then and I'd vote for just ignoring that parameter when it doesn't make sense (that is when dealing with symmetric keys) because really what you want is a key you can use.
So you would ask for the private key if you want to signing something. The public key if you want to verify a signature and so on. For symmetric keys, the key will always work (signing, verifying, ..) so you would get the behaviour you expected.

**Comment by jschlyter** [comment]
Ok, now `get_key()` works for all keys. If you try to do `get_key(private=True)` on a public key, it will raise ValueError though.

**Comment by rohe** [comment]
Perfect !

**Pull request merged**